rust/monovertex/src/server_info.rs

```diff
@@ -105,16 +143,19 @@ fn check_numaflow_compatibility(
         return Err(Error::ServerInfoError("invalid version".to_string()));
     }
 
+    // Strip the 'v' prefix if present.
+    let numaflow_version_stripped = numaflow_version.trim_start_matches('v');
+
     // Parse the provided numaflow version as a semantic version
-    let numaflow_version_semver = Version::parse(numaflow_version)
+    let numaflow_version_semver = Version::parse(numaflow_version_stripped)
         .map_err(|e| Error::ServerInfoError(format!("Error parsing Numaflow version: {}", e)))?;
 
     // Create a version constraint based on the minimum numaflow version
     let numaflow_constraint = format!(">={}", min_numaflow_version);
     check_constraint(&numaflow_version_semver, &numaflow_constraint).map_err(|e| {
         Error::ServerInfoError(format!(
             "numaflow version {} must be upgraded to at least {}, in order to work with current SDK version {}",
-            numaflow_version_semver, min_numaflow_version, e
+            numaflow_version_semver, human_readable(min_numaflow_version.to_string()), e
```

Comment:
why are you calling `to_string()`? just implement Debug for `Version` via [`NewType`](https://doc.rust-lang.org/rust-by-example/generics/new_types.html).

Reply:
@vigith , thanks for introducing to me the Debug trait. After learning a bit of it, I would like to keep my change as it is. My use case is to convert the version string used by constraint to a human-readable string. I don't want users to see "please update numaflow to 1.3.1-z", which is confusing, I want them to see "1-3-1".

This is a very simple string conversion. Using Debug will require me to create a new type which wraps a string and create object of that struct during compatibility check, which is extra code. Please let me know if my understanding is off.

Reply:
isn't "human-readable string" all about `Debug` Trait is?

Reply:
As discussed offline, I created a separate issue to track. https://github.com/numaproj/numaflow/issues/2076